WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs

```diff
@@ -1950,14 +1956,17 @@ SELECT id
                             }
                             case "ADD_LINK":
                             {
-                                // Check if the user requested this change to be synchronised.
-                                if (!entityTypeMergeSettings.Update)
+                                // check if the link type is in the list of changes 
+                                if (linkType != null)
                                 {
-                                    itemsProcessed++;
-                                    await UpdateProgressInQueue(databaseConnection, queueId, itemsProcessed);
-                                    continue;
+                                    if (!settings.LinkTypes.SingleOrDefault(s => s.Type == linkType && ( string.Equals(s.SourceEntityType, entityType) || string.Equals(s.DestinationEntityType, entityType))).Create)
+                                    {
+                                        itemsProcessed++;
+                                        await UpdateProgressInQueue(databaseConnection, queueId, itemsProcessed);
+                                        continue;
+                                    }
                                 }
```

Comment:
Please combine these if-statements since if the outer one returns true it only validates the inner one.

Reply:
Done.